### ISSUE TITLE
Login to Okta hangs

### ISSUE THREAD

**Issue by noahlz** [Versent/saml2aws#675]
I'm trying to login with Okta as my Idp. It hangs forever.

Here is verbose output:

```
$ saml2aws login -a prod --verbose
DEBU[0000] Running                                       command=login
DEBU[0000] check if Creds Exist                          command=login
DEBU[0000] Expand                                        name=/home/noahlz/.aws/credentials pkg=awsconfig
DEBU[0000] resolveSymlink                                name=/home/noahlz/.aws/credentials pkg=awsconfig
DEBU[0000] ensureConfigExists                            filename=/home/noahlz/.aws/credentials pkg=awsconfig
Using IDP Account prod to access Okta https://<myorg>.okta.com/home/amazon_aws/...
```

Here is the verbose curl output:

```
< HTTP/1.1 302 Found
< Date: Sun, 06 Jun 2021 04:22:17 GMT
< Content-Length: 0
< Connection: keep-alive
* Server nginx is not blacklisted
< Server: nginx
< Public-Key-Pins-Report-Only: pin-sha256="..."
< x-okta-request-id: ....
< x-xss-protection: 0
< p3p: CP="HONK"
< location: https://<myorg>.okta.com/login/login.htm?fromURI=%2Fhome%2Famazon_aws%....%2F...
< content-language: en
< Strict-Transport-Security: max-age=315360000; includeSubDomains
< X-Robots-Tag: none
< set-cookie: sid=""; Expires=Thu, 01-Jan-1970 00:00:10 GMT; Path=/
<
* Connection #0 to host <myorg>.okta.com left intact

```

What more information can I provide to resolve this issue?  I updated to the latest release (2.30.0).

**Comment by noahlz** [comment]
I also tried this with DUMP_CONTENT enabled:

```
$ DUMP_CONTENT=true saml2aws login -a prod --verbose
DEBU[0000] Running                                       command=login
DEBU[0000] check if Creds Exist                          command=login
DEBU[0000] Expand                                        name=/home/noahlz/.aws/credentials pkg=awsconfig
DEBU[0000] resolveSymlink                                name=/home/noahlz/.aws/credentials pkg=awsconfig
DEBU[0000] ensureConfigExists                            filename=/home/noahlz/aws/credentials pkg=awsconfig
Using IDP Account alpha-prod to access Okta https://<myorg>.okta.com/home/amazon_aws/.../...
```

It also hangs at this point.

**Comment by noahlz** [comment]
Thank you to this StackOverflow post, I got a full trace from go with GOTRACEBACK=all

```
SIGQUIT: quit
PC=0x46dce0 m=0 sigcode=0

goroutine 0 [idle]:
runtime.epollwait(0x7ffc00000004, 0x7ffcdbca3450, 0xffffffff00000080, 0x0, 0xffffffff, 0x0, 0x0, 0x1071c80, 0x2, 0x0, ...)
        runtime/sys_linux_amd64.s:725 +0x20
runtime.netpoll(0xffffffffffffffff, 0x0)
        runtime/netpoll_epoll.go:126 +0x92
runtime.findrunnable(0xc00003b000, 0x0)
        runtime/proc.go:2879 +0x3ee
runtime.schedule()
        runtime/proc.go:3125 +0x2d7
runtime.park_m(0xc000000180)
        runtime/proc.go:3274 +0x9d
runtime.mcall(0x0)
        runtime/asm_amd64.s:327 +0x5b

goroutine 1 [chan receive]:
github.com/gsterjov/go-libsecret.(*Prompt).Prompt(0xc00021f1e8, 0x0, 0x0, 0x0)
        github.com/gsterjov/go-libsecret@v0.0.0-20161001094733-a6f4afe4910c/prompt.go:50 +0x1b3
github.com/gsterjov/go-libsecret.(*Service).Unlock(0xc0000ae390, 0xd48aa0, 0xc000119e78, 0xc000119e78, 0x1)
        github.com/gsterjov/go-libsecret@v0.0.0-20161001094733-a6f4afe4910c/service.go:112 +0x2d9
github.com/99designs/keyring.(*secretsKeyring).Get(0xc0000a0360, 0xc0002b23c1, 0x47, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        github.com/99designs/keyring@v1.1.6/libsecret.go:127 +0x3ec
github.com/versent/saml2aws/v2/helper/linuxkeyring.(*KeyringHelper).Get(0xc0002d6570, 0xc0002b23c1, 0x47, 0xc14230, 0xc00021f730, 0x630c85, 0xc00010c0a0, 0x2, 0xc000140380)
        github.com/versent/saml2aws/v2/helper/linuxkeyring/linuxkeyring.go:56 +0x87
github.com/versent/saml2aws/v2/helper/credentials.LookupCredentials(0xc00017ec80, 0xc000117a4b, 0x4, 0x3, 0x3)
        github.com/versent/saml2aws/v2/helper/credentials/saml.go:12 +0x50
github.com/versent/saml2aws/v2/cmd/saml2aws/commands.resolveLoginDetails(0xc0002f2640, 0xc000108f40, 0x0, 0x0, 0x1)
        github.com/versent/saml2aws/v2/cmd/saml2aws/commands/login.go:184 +0x418
github.com/versent/saml2aws/v2/cmd/saml2aws/commands.Login(0xc000108f40, 0xbd1098, 0x5)
        github.com/versent/saml2aws/v2/cmd/saml2aws/commands/login.go:69 +0x233
main.main()
        command-line-arguments/main.go:183 +0x53e5

goroutine 19 [IO wait]:
internal/poll.runtime_pollWait(0x7f3c9f04c348, 0x72, 0x1000)
        runtime/netpoll.go:222 +0x55
internal/poll.(*pollDesc).wait(0xc00017e218, 0x72, 0x0, 0x10, 0xc00038ec20)
        internal/poll/fd_poll_runtime.go:87 +0x45
internal/poll.(*pollDesc).waitRead(...)
        internal/poll/fd_poll_runtime.go:92
internal/poll.(*FD).ReadMsg(0xc00017e200, 0xc000384e70, 0x10, 0x10, 0xc00038ec20, 0x1000, 0x1000, 0x0, 0x0, 0x0, ...)
        internal/poll/fd_unix.go:249 +0x252
net.(*netFD).readMsg(0xc00017e200, 0xc000384e70, 0x10, 0x10, 0xc00038ec20, 0x1000, 0x1000, 0xea00000008000103, 0x0, 0xc0002f7b00, ...)
        net/fd_posix.go:67 +0x90
net.(*UnixConn).readMsg(0xc000102158, 0xc000384e70, 0x10, 0x10, 0xc00038ec20, 0x1000, 0x1000, 0xc00039c050, 0xc0002efe90, 0x10f1290, ...)
        net/unixsock_posix.go:115 +0x91
net.(*UnixConn).ReadMsgUnix(0xc000102158, 0xc000384e70, 0x10, 0x10, 0xc00038ec20, 0x1000, 0x1000, 0xc00039a330, 0xc00008dc88, 0x40d9bb, ...)
        net/unixsock.go:143 +0x9d
github.com/godbus/dbus.(*oobReader).Read(0xc00038ec00, 0xc000384e70, 0x10, 0x10, 0xc00008dcb8, 0x40e118, 0x1020)
        github.com/godbus/dbus@v4.1.0+incompatible/transport_unix.go:21 +0x8d
io.ReadAtLeast(0xd47880, 0xc00038ec00, 0xc000384e70, 0x10, 0x10, 0x10, 0xc00008dd00, 0x4644e0, 0xc000000180)
        io/io.go:328 +0x87io.ReadFull(...)
        io/io.go:347
github.com/godbus/dbus.(*unixTransport).ReadMessage(0xc00010bd60, 0xc0002b52c0, 0xa, 0xc0002bc8e8)
        github.com/godbus/dbus@v4.1.0+incompatible/transport_unix.go:85 +0x12e
github.com/godbus/dbus.(*Conn).inWorker(0xc0002ce000)
        github.com/godbus/dbus@v4.1.0+incompatible/conn.go:285 +0x57
created by github.com/godbus/dbus.(*Conn).Auth
        github.com/godbus/dbus@v4.1.0+incompatible/auth.go:118 +0x667

goroutine 20 [chan receive]:
github.com/godbus/dbus.(*Conn).outWorker(0xc0002ce000)
        github.com/godbus/dbus@v4.1.0+incompatible/conn.go:427 +0x55
created by github.com/godbus/dbus.(*Conn).Auth
        github.com/godbus/dbus@v4.1.0+incompatible/auth.go:119 +0x68c

rax    0xfffffffffffffffc
rbx    0xffffffff
rcx    0xffffffffffffffff
rdx    0x80
rdi    0x4
rsi    0x7ffcdbca3450
rbp    0x7ffcdbca3a50
rsp    0x7ffcdbca3400
r8     0x0
r9     0x7
r10    0xffffffff
r11    0x246
r12    0x0
r13    0x0
r14    0xc130a8
r15    0x0
rip    0x46dce0
rflags 0x246
cs     0x33
fs     0x0
gs     0x0

```

**Comment by noahlz** [comment]
My reading of this stack trace is that it's related to  gsterjov/go-libsecret and Go is hanging trying to interact with the  freedesktop.org Secret Service DBus API

**Comment by noahlz** [comment]
Ok, I solved this issue...

Thanks to the stack trace, I was able to determine that saml2aws was blocking on obtaining a response from the secret service dbus, i.e. so it can get username/password input securely. However, I am working remotely (thank you, covid) from the command line ssh'ing into a remote Linux machine.

This Linux machine (ubuntu) is not headless, and has a Gnome desktop. That desktop was presenting the "keyring" popup, prompting for a password.  saml2aws (via go-libsecret) was blocked waiting for that popup to be closed.

After some googling for "disable keyring popup ubuntu" I found this post which describes some steps that work.

https://ubuntuforums.org/showthread.php?t=1655397&p=10301640#post10301640

Recommendation: provide some entries in DUMP_CONTENT that provides a hint to the user what is happening. In this case, prompting for username/password.
